src/ui/components/Overlay/index.js

```diff
@@ -1,80 +1,110 @@
+/* @flow */
+import invariant from 'invariant';
 import makeClassName from 'classnames';
 import * as React from 'react';
-import PropTypes from 'prop-types';
+import { compose } from 'redux';
+
+import withUIState from 'core/withUIState';
 
 import './styles.scss';
 
-export default class Overlay extends React.Component {
-  static propTypes = {
-    children: PropTypes.node,
-    className: PropTypes.string,
-    onEscapeOverlay: PropTypes.func,
-    visibleOnLoad: PropTypes.bool,
-  };
+// TODO: look into this as I don't see this working on the frontend
+// even in other envs.
+export const onClickBackground = (
+  event: SyntheticEvent<HTMLButtonElement>,
+  // eslint-disable-next-line no-use-before-define
+  _this: OverlayBase,
+) => {
+  if (_this.props.onEscapeOverlay) {
+    _this.props.onEscapeOverlay();
+  }
+  _this.hide();
+};
+
+type Props = {|
+  className?: string,
+  children: React.Element<any>,
+  id: string,
+  onEscapeOverlay?: Function,
```

Comment:
If you make this `onEscapeOverlay?: () => void` then Flow will validate the function signature. It's more fine-grained than `Function`; it says it's a function that takes no arguments.